src/vm/perfmap.cpp

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// ===========================================================================
+// File: perfmap.cpp
+//
+
+#include "common.h"
+#include "perfmap.h"
+
+PerfMap * PerfMap::s_Current = NULL;
+
+// Initialize the map for the process - called from EEStartupHelper.
+void PerfMap::Initialize()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    // Only enable the map if requested.
+    if (CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_PerfMapEnabled))
+    {
+        // Get the current process id.
+        int currentPid = GetCurrentProcessId();
+
+        // Create the map.
+        s_Current = new PerfMap(currentPid);
+    }
+}
+
+// Log a method to the map.
+void PerfMap::LogMethod(MethodDesc * pMethod, PCODE pCode, size_t codeSize)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    if (s_Current != NULL)
+    {
+        s_Current->Log(pMethod, pCode, codeSize);
+    }
+}
+
+// Destroy the map for the process - called from EEShutdownHelper.
+void PerfMap::Destroy()
+{
+    if (s_Current != NULL)
+    {
+        delete s_Current;
+        s_Current = NULL;
+    }
+}
+
+// Construct a new map for the process.
+PerfMap::PerfMap(int pid)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    // Initialize with no failures.
+    m_ErrorEncountered = false;
+
+    // Build the path to the map file on disk.
+    SString path;
+#ifdef FEATURE_PAL
+    path.Append(W("/tmp/perf-"));
+#else
+    path.Append(W("C:\\perf-"));
+#endif
+    
+    WCHAR strPid[10];
+    _itow_s(
+        pid,    // source
+        strPid, // dest
+        10,     // size in characters
+        10);    // radix
+
+    path.Append((WCHAR *)&strPid);
```

Comment:
It seems it would be easier to read if you used the SString::Printf method for the whole file name / path formatting.